R/nowcast_df-class.R

```diff
@@ -0,0 +1,30 @@
+#' Nowcast Data.frame Object
+#' @name nowcast_df-class
+#' @aliases nowcast_df
+#' @family nowcast_df
+#'
+#' @description
+#' A `nowcast_df` object which contains point or probabilistic nowcasts
+#'   alongside reference dates and any additional metadata, in tidy data format.
+#'   Nowcasts are presented aggregated across delays, by reference date.
+#'
+#' @section Structure:
+#' A `nowcast_df` is a data.frame with the following columns:
+#' \describe{
+#'  \item{reporting_triangle_matrix}{Matrix with which rows are reference
+#'    times and columns are delays and
+#'    entries are incident cases at each reference time and delay.}
+#'  \item{reference_date}{Dates corresponding to the reference times of the
+#'  nowcast.}
+#'  \item{time}{Integer indicating the time index}
+#'  \item{pred_count}{Numeric indicating the estimated total counts aggregated
+#'    across delays at each reference date.}
+#'  \item{draw}{Integer indexing the sample from the probabilistic nowcast
```

Comment:
you could save yourself some future pain in plotting etc but having this always be there but being 1 for point

Reply:
A bit semantics but 1 draw from the observation model is different from the point estimate, so I feel like this might be misleading (e.g. in the frowned upon case where a user wants 1 probabilistic draw, the output would like identical to the point estimate in terms of metadata structure but it would not be).

We could add a column `output_type` and have that be `point` or `samples` if we want to do this, then for `draw` either have 1 or NULL for the point estimate.